### PR TITLE
exercises(triangle): simplify error set

### PR DESCRIPTION
**Pull request by booniepepper**
I took an editing pass of the triangle exercise

* https://github.com/exercism/zig/issues/215 Replaced the `TriangleError.InvalidEquality` with `TriangleError.Invalid`
* ~~https://github.com/exercism/zig/issues/216 Removed private `fn` stubs in the initial code~~ @ee7 is too fast lol
* Misc small refactors to the example

Closes: #215

### CODE REVIEW

File: exercises/practice/triangle/.meta/example.zig [hunk 1]
```diff
@@ -1,6 +1,6 @@
 pub const TriangleError = error{
     Degenerate,
-    InvalidInequality,
+    Invalid,
 };
```
Comment (hiljusti) [inline exercises/practice/triangle/.meta/example.zig:4]: Do other tracks even make this distinction?

Comment (hiljusti) [reply exercises/practice/triangle/.meta/example.zig:4]: Note to myself:

Degenerate = Zero area, the "triangle" is actually a line with 3 points on it

Invalid = Can't be a triangle

(At least, on a flat Euclidean plane)

**Comment by ErikSchierboom** [comment]
> which is an argument for just one error name. Otherwise the user has to understand when to return Degenerate, without instruction. If we keep the distinction, I think we should explain in an instructions_append.md.

I think I would prefer just one error name. As your research shows, we removed degenerate from the instructions, so my preference would be to not add it here.

File: exercises/practice/triangle/.meta/example.zig (hunk 1 above)
Comment (hiljusti) [reply exercises/practice/triangle/.meta/example.zig:4]: Thanks for the deep dive!

**Pull request merged**